<?xml version="1.0"?>
<launch>
    <!--Input arguments-->
    <arg name="paused" default="False"/>
    <arg name="use_sim_time" default="true"/>
    <arg name="gui" default="true"/>
    <arg name="headless" default="false"/>
    <arg name="debug" default="false"/>

    <!-- Spawn worlds-->
    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="world_name" value="$(find simulation)/worlds/flat.world"/>
        <arg name="debug" value="$(arg debug)"/>
        <arg name="gui" value="$(arg gui)"/>
        <arg name="paused" value="$(arg paused)"/>
        <arg name="use_sim_time" value="$(arg use_sim_time)"/>
        <arg name="headless" value="$(arg headless)"/>
    </include>

    <node pkg="tf" type="static_transform_publisher" name="link1_broadcaster" args="0 0 0 0 0 0 1  /world /map 100" />
    <node pkg="tf" type="static_transform_publisher" name="link2_broadcaster" args="0 0 0 0 0 0 1 /map /odom 100" />

    <node name="env_gen_services" pkg="simulation" type="env_gen_services.py" output="screen"/>
    <node name="robot_spawn_service" pkg="simulation" type="robot_spawn_service.py" output="screen"/>

</launch>